projects/api/src/category/category.service.ts

```diff
@@ -0,0 +1,80 @@
+import { BadRequestException, Injectable, NotFoundException } from '@nestjs/common';
+
+import PrismaService from '@/prisma/prisma.service';
+
+import CreateCategoryDto from './dto/create-category.dto';
+import UpdateCategoryDto from './dto/update-category.dto';
+import Category from './entities/category.entity';
+
+@Injectable()
+class CategoryService {
+  constructor(private readonly prisma: PrismaService) {}
+
+  async create(createCategoryDto: CreateCategoryDto): Promise<Category> {
+    try {
+      const category: Category = await this.prisma.category.create({
+        data: createCategoryDto,
+      });
+
+      return category;
+    } catch (error) {
+      if (error?.meta?.target?.includes('name')) {
+        throw new BadRequestException('Category name already exists');
```

Comment:
If in your model you don't put that the name field is unique, you don't need to validate this possible error, but it is necessary so keep that but change the model.